File: java/base/src/main/vsdk/toolkit/environment/geometry/geometricProcessing/polyhedralBoundedSolidOperators/construction/_PolyhedralBoundedSolidProfileDifferenceFallbackSpec.java
package vsdk.toolkit.environment.geometry.geometricProcessing.polyhedralBoundedSolidOperators;

import java.util.ArrayList;

import vsdk.toolkit.common.linealAlgebra.Vector3Dd;

final class _PolyhedralBoundedSolidProfileDifferenceFallbackSpec
{
    final ArrayList<Vector3Dd> clippedProfileAtCut;
    final double xCut;
    final double xMax;
    final double[] minuendBounds;

    _PolyhedralBoundedSolidProfileDifferenceFallbackSpec(
        ArrayList<Vector3Dd> clippedProfileAtCut,
        double xCut,
        double xMax,
        double[] minuendBounds)
    {
        this.clippedProfileAtCut = clippedProfileAtCut;
        this.xCut = xCut;
        this.xMax = xMax;
        this.minuendBounds = minuendBounds;
    }
}

File: typescript/base/src/main/vsdk/toolkit/environment/geometry/geometricProcessing/polyhedralBoundedSolidOperators/construction/_PolyhedralBoundedSolidProfileDifferenceFallbackSpec.ts
import { Vector3Dd } from "../../../../../common/linealAlgebra/Vector3Dd.js";

export class _PolyhedralBoundedSolidProfileDifferenceFallbackSpec {
    public constructor(
        public readonly clippedProfileAtCut: Vector3Dd[],
        public readonly xCut: number,
        public readonly xMax: number,
        public readonly minuendBounds: number[],
    ) {}
}
